crates/sui-indexer-alt/src/handlers/coin_balance_buckets_pruner.rs

```diff
@@ -39,14 +40,29 @@ impl Handler for CoinBalanceBucketsPruner {
             let cp = to_prune.entry(object_id).or_default();
             *cp = std::cmp::max(*cp, cp_sequence_number_exclusive);
         }
-        let mut committed_rows = 0;
-        for (object_id, cp_sequence_number_exclusive) in to_prune {
-            committed_rows += diesel::delete(coin_balance_buckets::table)
-                .filter(coin_balance_buckets::object_id.eq(object_id.as_slice()))
-                .filter(coin_balance_buckets::cp_sequence_number.lt(cp_sequence_number_exclusive))
-                .execute(conn)
-                .await?;
-        }
-        Ok(committed_rows)
+        let values = to_prune
+            .iter()
+            .map(|(object_id, seq_number)| {
+                let object_id_hex = hex::encode(object_id);
+                format!("('\\x{}'::BYTEA, {}::BIGINT)", object_id_hex, seq_number)
+            })
+            .collect::<Vec<_>>()
+            .join(",");
+        let query = format!(
+            "
+            WITH to_prune_data (object_id, cp_sequence_number_exclusive) AS (
+                VALUES {}
+            )
+            DELETE FROM coin_balance_buckets
+            USING to_prune_data
+            WHERE coin_balance_buckets.{:?} = to_prune_data.object_id
+              AND coin_balance_buckets.{:?} < to_prune_data.cp_sequence_number_exclusive
+            ",
+            values,
+            dsl::object_id,
+            dsl::cp_sequence_number,
+        );
+        let rows_deleted = sql_query(query).execute(conn).await?;
```

Comment:
nit: inline?

Reply:
hmm I prefer to keep them in two lines for clarity, hope that's ok!